### ISSUE TITLE
tf.keras.ops.numpy.sinc returns NaN gradient at zero where derivative should be zero

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#127226]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0, also reproduced on tf 2.22.0-dev20260904

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.13.5

### Current behavior?

`tf.keras.ops.numpy.sinc` returns `nan` for the first derivative at a point where the normalized sinc function has a smooth extension and the derivative should be finite.

In the reproducer below, the function evaluates `tf.keras.ops.numpy.sinc(2 * t + 1)` at `t = -0.5`, so the argument to `sinc` is exactly zero. The forward value is correctly returned as `1.0`. However, TensorFlow autodiff returns `nan` for the derivative with respect to `t`.

The expected derivative is `0.0`, including the chain rule through `2 * t + 1`.

Expected behavior?

The first derivative of `tf.keras.ops.numpy.sinc(2 * t + 1)` at `t = -0.5` should be `0.0`, not `nan`.

### Standalone code to reproduce the issue

```shell
import os
os.environ["CUDA_VISIBLE_DEVICES"] = ""
os.environ["OMP_NUM_THREADS"] = "1"
os.environ["TF_NUM_INTRAOP_THREADS"] = "1"
os.environ["TF_NUM_INTEROP_THREADS"] = "1"

import tensorflow as tf

def target(t):
    return tf.keras.ops.numpy.sinc(2 * t + 1)

x = tf.constant(-0.5, dtype=tf.float64)

with tf.GradientTape() as tape:
    tape.watch(x)
    y = target(x)

actual = tape.gradient(y, x)
expected = 0.0

print("forward:", y.numpy())
print("actual:", actual.numpy())
print("expected:", expected)

if tf.math.is_nan(actual):
    print("BUG REPRODUCED: tf.keras.ops.numpy.sinc returns NaN gradient at zero")
else:
    print("not reproduced")
```

### Relevant log output

```shell
forward: 1.0
actual: nan
expected: 0.0
BUG REPRODUCED: tf.keras.ops.numpy.sinc returns NaN gradient at zero
```

**Comment by Kayyuri** [comment]
Hi @ALinrunrun, Thank you for reporting this issue!

You've clearly demonstrated that  `tf.keras.ops.numpy.sinc` returns a `NaN` gradient when its input is exactly zero, despite the mathematical derivative at that point being `0.0`. 

We’ll investigate the issue on our end and provide an update. Thanks!

**Comment by Venkat6871** [comment]
Hi @ALinrunrun ,

Thanks for raising this issue. I have opened a [PR](https://github.com/tensorflow/tensorflow/pull/127478) with a fix for the reported behavior. Once the PR is merged, this issue should be resolved.

Thanks!

**Comment by OssAgenticsT** [comment]
Closing this issue as resolved. Feel free to open a new issue or comment here if you need further assistance!

**Comment by Phoebus-Liu** [comment]
Hi @OssAgenticsT,

Just a note so this doesn't get closed prematurely: the fix for this issue (PR #127742) has been approved but is **not merged yet**, so the behavior on `master` is still unchanged.

I'd suggest keeping this open until #127742 lands — it carries `Fixes #127226`, so it will be closed automatically at that point.

Thanks!
